frontend/src/metabase/query_builder/components/filters/BulkFilterModal/NumberFilterEditor/NumberFilterEditor.tsx

```diff
@@ -0,0 +1,132 @@
+import { useEffect } from "react";
+import { usePrevious } from "react-use";
+import { t } from "ttag";
+import _ from "underscore";
+import { Flex, NumberInput, Text } from "metabase/ui";
+import { ColumnValuesWidget } from "metabase/common/components/ColumnValuesWidget";
+import { useNumberFilter } from "metabase/common/hooks/filters/use-number-filter";
+import * as Lib from "metabase-lib";
+import type { FilterPickerWidgetProps } from "../types";
+import { FilterOperatorPicker } from "../FilterOperatorPicker";
+
+export function NumberFilterEditor({
+  query,
+  stageIndex,
+  column,
+  filter,
+  onChange,
+}: FilterPickerWidgetProps) {
+  const isID = Lib.isPrimaryKey(column) || Lib.isForeignKey(column);
+  const defaultOperator = isID ? "=" : "between";
+
+  const {
+    operator,
+    values,
+    valueCount,
+    hasMultipleValues,
+    availableOperators,
+    setOperator,
+    setValues,
+    getFilterClause,
+  } = useNumberFilter({
+    query,
+    stageIndex,
+    column,
+    filter,
+    defaultOperator,
+  });
+
+  const previousOperator = usePrevious(operator);
+  const previousValues = usePrevious(values);
+
+  useEffect(() => {
+    if (operator !== previousOperator || !_.isEqual(values, previousValues)) {
+      onChange(getFilterClause());
+    }
+  });
+
+  return (
+    <Flex align="center">
+      <FilterOperatorPicker
+        value={operator}
+        options={availableOperators}
+        onChange={setOperator}
+      />
+      <NumberValueInput
+        column={column}
+        values={values}
+        valueCount={valueCount}
+        hasMultipleValues={hasMultipleValues}
+        onChange={setValues}
+      />
+    </Flex>
+  );
+}
+
+type NumberValue = number | "";
+
+interface NumberValueInputProps {
+  column: Lib.ColumnMetadata;
+  values: NumberValue[];
+  valueCount: number;
+  hasMultipleValues?: boolean;
+  onChange: (values: NumberValue[]) => void;
+}
+
+function NumberValueInput({
```

Comment:
Is it different to what we have in `NumberFilterPicker`?

Reply:
Not yet, I just copy-pasted it for now. It should diverge eventually